taskcluster/client.py

```diff
@@ -67,6 +75,17 @@ def __init__(self, options=None, session=None):
                     except:
                         s = '%s (%s) must be unicode encodable' % (x, credentials[x])
                         raise exceptions.TaskclusterAuthFailure(s)
+
+        baseUrl = o.get('baseUrl')
+        rootUrl = o.get('rootUrl')
+
+        if rootUrl and not o.get('version'):
+            s = 'specifying a rootUrl requires a version in the API reference'
+            raise exceptions.TaskclusterFailure(s)
+        elif rootUrl:
+            name = o.get('name', urllib.parse.urlparse(baseUrl).netloc.split('.', 1)[0])
+            o['baseUrl'] = liburl.api(rootUrl, name, 'v' + o['version'], '')
```

Comment:
As @imbstack indicated, this is wrong -- `o['version'] is the version of the reference format.  The [schema](https://github.com/taskcluster/taskcluster-references/blob/master/schemas/api-reference-v0.yml#L81) enforces that it is always `0`.

Reply:
For example, it's `0` for [secrets](http://references.taskcluster.net/secrets/v1/api.json)

Reply:
How should we figure out which API version?  Do we default to a static `1` until we figure out how to specify that data?  I didn't see anything like apiVersion in the schema.

Reply:
It's in the manifest, not in the API schema.  So during parsing of the manifest we'll need to extract it.

Reply:
But we don't use manifests everywhere.  An API reference is a description of an interface to a service, the version of that interface is an important part of that description.

Reply:
It is [there](https://github.com/taskcluster/taskcluster-lib-api/blob/master/src/schemas/api-reference.json), too.  Nobody has figured out how to synchronize those yet.

Reply:
https://bugzilla.mozilla.org/show_bug.cgi?id=1145220 is part of figuring that out.